### PR TITLE
Hetzner: Skip operating system validation if custom image is provided

### PR DESCRIPTION
**Pull request by P4sca1**
This should make it possible to use operating systems on Hetzner Cloud, that are not natively supported, by using custom images (snapshots).

A side-effect of this commit is, that images returned by `getNameForOs()` are now also validated against the Hetzner API.

For reference: #1817

Requirements to use custom operating systems:
1. Create a snapshot of the operating system you want to use on Hetzner Cloud, e.g. using https://github.com/apricote/hcloud-upload-image/
2. Add a new `OperatingSystemProfile`
3. Set the image in `MachineDeployment` to the name of the snapshot created in step 1

/kind bug

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
NONE
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: pkg/cloudprovider/provider/hetzner/provider.go [hunk 1]
```diff
@@ -223,13 +218,19 @@ func (p *provider) Validate(ctx context.Context, _ *zap.SugaredLogger, spec clus
 		}
 	}
 
-	if c.Image != "" {
-		//nolint:staticcheck // We do not have the architecture available here.
-		if _, _, err = client.Image.Get(ctx, c.Image); err != nil {
-			return fmt.Errorf("failed to get image: %w", err)
+	image := c.Image
+	if image == "" {
+		image, err = getNameForOS(pc.OperatingSystem)
```
Comment (kron4eg) [inline pkg/cloudprovider/provider/hetzner/provider.go:223]: This `image` variable is not used later.

Comment (P4sca1) [reply pkg/cloudprovider/provider/hetzner/provider.go:223]: You are right, I pushed a new commit.

**Comment by kron4eg** [comment]
@P4sca1 hey, thanks for contribution. Please add a [Signed-off-by](https://github.com/kubermatic/machine-controller/blob/main/CONTRIBUTING.md#certificate-of-origin) to your commit (for example with `git commit --amend --signoff`).

**Comment by P4sca1** [comment]
I edited the file via the GitHub mobile app, which unfortunately does not sign the commits.
I force-pushed a single commit to my patch branch, which is signed and already contains the fix for the image variable.

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve

**Pull request merged**